## Table of Content

1. **[Conditional Statements](#cond)**
2. **[Python Flow Control](#flow_control)**
3. **[List Comprehension](#lc)**



<a id="cond"> </a>
## 1. Conditional Statements

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. Write a program to check whether a given number is multiple of 7 (use if statement)</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [3]:
num = int(input("enter your number- "))
          
if num%7==0:
          print("num is divisible by 7")

enter your number- 7
num is divisible by 7


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>2. Write a code to print the maximum of two numbers (use if-else)</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [3]:
num1 = int(input("enter your number- "))
num2 = int(input("enter your number- "))

if num1>num2:
    print("num1 is greater than num2")
else:
    print("num2 is greater than num1")

enter your number- 32
enter your number- 44
num2 is greater than num1


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. Write a program to check whether a number is positive, negative or zero (use if-elif) </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [5]:
num = int(input("enter your number- "))

if num>0:
    print("number is positive")
elif num<=0:
        print("number is negative or zero")

enter your number- 7
number is positive


### Let's begin with some hands-on practice exercises

<a id= "flow_control"> </a>
## 2. Python Flow Control

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. Write a code to print table of 5 using assignment operand +=.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [52]:
n = 5

for i in range(1,11):

    print(5,"x",i ,"=" ,5*i)

5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. Write a program for generating a fibonacci series starting with 0 and 1 for the next 10 values using a while loop. </b>
                </font>
            </div>
        </td>
    </tr>

</table>

In [4]:
a = 0
b = 1
n = 10
i = 0
print(a)
print(b)
while i<=n:
    c = a+b
    a=b
    b=c
    i+=1
    print(c)

0
1
1
2
3
5
8
13
21
34
55
89
144


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>6. Write a program to check whether a number is greater than the other using if statement (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [8]:
num1 = int(input("enter your number- "))
num2 = int(input("enter your number- "))

if num1>num2:
    print(num1, "is greater than num2")
else:
    print(num2, "is greater than num1")

enter your number- 24
enter your number- 23
24 is greater than num2


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>7. Write a code to check whether a number is divisible by 7 or not (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [10]:
num = float(input("enter your number- "))

if num%7==0:
    print("divisible by 7 ")
else:
    print("not divisible by 7")
    

enter your number- 35
divisible by 7 


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>8. Write a code to find factorial of a number (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [12]:
num = int(input("enter a number- "))
factorial = 1
for i in range(1,num+1):
    factorial = factorial * i
print("The factorial of",num,"is",factorial)

enter a number- 5
The factorial of 5 is 120


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>9. Write a program to check whether a number is prime or not (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [20]:
num = int(input("enter a number- "))
if num>1:
    for i in range(2,1):
    if num%i==0:
        print("not prime")
        break
else:
    print("prime")

IndentationError: expected an indented block (1651373271.py, line 4)

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>10. Write a program to check whether two numbers are amicable or not (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [24]:
a = int(input("enter a number- "))
b= int(input("enter a number- "))

sum_a = 0
sum_b = 0

for i in range(1,a):
    if a%i == 0:
        sum_a = sum_a +1
for j in range(1,b):
    if b%i==0:
        sum_b = sum_b+1
if sum_a == b and sum_b == a:
    print("amicable")
else:
        print("not")

enter a number- 12
enter a number- 12
not


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>11. Reverse string using a for loop (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [27]:
str = input("Enter the value")
str_rev = " "
for i in str:
    str_rev = i + str_rev
print(str_rev)

Enter the value24
42 


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>12. Write a code to find the average of given numbers (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [29]:
num = int(input("enter the number of values to check average "))
sum_num = 0
for i in range(num):
    val = float(input("Enter the value "))
    sum_num = sum_num + val
print("The average pf given values is ", sum_num/num)

enter the number of values to check average 1
Enter the value 2
The average pf given values is  2.0



<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>13. Write a program to find the area of a circle for a given radius (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>


In [26]:
radius = int(input("Enter radius-"))

print("area of circle =",3.14*radius**2)

Enter radius-6
area of circle = 113.04


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>14. Write a code to find the simple interest (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [30]:
# si = ptr/100
P = int(input("Enter the principal amount in rs "))
T = float(input("enter the time in years "))
R = float(input("enter the interest value in % "))
SI = (P*T*R)/100
print("Simple interest for given inputs in rupees is",SI)

Enter the principal amount in rs 3000
enter the time in years 5
enter the interest value in % 10
Simple interest for given inputs in rupees is 1500.0


<a id= "lc"> </a>
## 3. List Comprehension

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>15. Use list comprehension to obtain the squre root of first 10 natural numbers.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [6]:
x = [1,2,3,4,5,6,7,8,9,10]

blank_list = []

blank_list= [i**1/2 for i in x ]
print(blank_list)

[0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>16. Use list comprehension to find the used vowels in a given sentence and/or word (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [1]:
sentence = input("enter your sentence")
vowels ="aeiou"

blank_list =[]

blank_list = [i for i in sentence if i in vowels]
print(blank_list)

enter your sentencesanjeev is
['a', 'e', 'e', 'i']


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>17. Create a dictionary and access it values using a condition on its key. The data is given below. Let the condidtion on the key is that it should be a even number.</b>
                </font>
            </div>
       </td>
        
    </tr>
</table>


<table>
<tr>
    <th>Key</th>
    <td>1</td>
    <td>2</td>
    <td>3</td>
    <td>4</td> 
    <td>5</td> 
</tr>
<tr>
    <th>Name</th>
    <td>Aman</td>
    <td>Mohit</td>
    <td>Guari</td>
    <td>Imran</td>
    <td>Roma</td
     </tr>
<tr>
    <th>Marks</th>
    <td>24</td>
    <td>25</td>
    <td>26</td>
    <td>24</td>
    <td>27</td>
   </tr>
</table>

In [3]:
dict = {1:{"name":"Aman","marks":24}\
        ,2:{"name":"Mohit","marks":25}\
        ,3:{"name":"Gurai","marks":26}\
        ,4:{"name":"Imran","marks":24}\
        ,5:{"name":"Imran","marks":27}}

In [4]:
for i in dict:
    if i%2==0:
        print(dict[i])

{'name': 'Mohit', 'marks': 25}
{'name': 'Imran', 'marks': 24}


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>18. Use list comprehension to find even and odd numbers from first 20 whole numbers.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [8]:
x = range(0,20)

b=[]
c=[]
b = [i for i in x if i%2==0]
c=[i for i in x if i%2!=0]
print(b)
print(c)

[0, 2, 4, 6, 8, 10, 12, 14, 16, 18]
[1, 3, 5, 7, 9, 11, 13, 15, 17, 19]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>19. Use list comprehension to print numbers divisible  by 2 and 3 in between 1 and 100. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [10]:
x = range(1,101)

b = []

b = [i for i in x if i%2==0 if i%3==0]
print(b)

[6, 12, 18, 24, 30, 36, 42, 48, 54, 60, 66, 72, 78, 84, 90, 96]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>20. Use list comprehension to create a dictionary such that its keys are numbers from 1 to 10 and values are the corresponding to that key are its cube.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [29]:
b = [(i,i**3) for i in range(1,11)]
print(b)

[(1, 1), (2, 8), (3, 27), (4, 64), (5, 125), (6, 216), (7, 343), (8, 512), (9, 729), (10, 1000)]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>21. Use list comprehension to extract numbers from a string </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [11]:
text = "current asset is 2300000 and current liability is 124500"

In [13]:
text.split()
print(text.split())

['current', 'asset', 'is', '2300000', 'and', 'current', 'liability', 'is', '124500']


In [14]:
b = []

In [15]:
b = [i for i in text.split() if i.isdigit()==True ]

In [16]:
print(b)

['2300000', '124500']


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>22. Use list comprehension to print table of 11 to 20 as shown below.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

<table>
<tr>
    <td>11</td><td>22</td><td>33</td><td>44</td><td>55</td><td>66</td><td>77</td><td>88</td><td>99</td><td>110</td> 
</tr>
    
<tr>
    <td>12</td><td>24</td><td>36</td><td>48</td><td>60</td><td>72</td><td>84</td><td>96</td><td>108</td><td>120</td> 
</tr>

<tr>
    <td>13</td><td>36</td><td>39</td><td>52</td><td>65</td><td>78</td><td>91</td><td>104</td><td>117</td><td>130</td> 
</tr> 

<tr>
     <td>14</td><td>28</td><td>42</td><td>56</td><td>70</td> <td>84</td><td>98</td><td>112</td><td>126</td><td>140</td> 
</tr>

<tr>
     <td>15</td><td>30</td><td>45</td><td>60</td><td>75</td><td>90</td><td>105</td><td>120</td><td>135</td><td>150</td> 
</tr>

<tr>
     <td>16</td><td>32</td><td>48</td><td>64</td><td>80</td><td>96</td><td>112</td><td>128</td><td>144</td><td>160</td> 
</tr> 

<tr>
     <td>17</td><td>34</td><td>51</td><td>68</td><td>85</td><td>102</td><td>119</td><td>136</td><td>153</td><td>170</td> 
</tr> 

<tr>
     <td>18</td><td>36</td><td>54</td><td>72</td><td>90</td> <td>108</td><td>126</td><td>144</td><td>162</td><td>180</td> 
</tr>

<tr>
 <td>19</td><td>38</td><td>57</td><td>76</td><td>95</td><td>114</td><td>133</td><td>152</td><td>171</td><td>190</td> 
</tr>

<tr>
     <td>20</td><td>40</td><td>60</td><td>80</td><td>100</td><td>120</td><td>140</td><td>160</td><td>180</td><td>200</td> 
<tr>    
    
</table>

In [38]:

b = [[i*n  for n in range(1,11)] for i in range(11,21)]
b

[[11, 22, 33, 44, 55, 66, 77, 88, 99, 110],
 [12, 24, 36, 48, 60, 72, 84, 96, 108, 120],
 [13, 26, 39, 52, 65, 78, 91, 104, 117, 130],
 [14, 28, 42, 56, 70, 84, 98, 112, 126, 140],
 [15, 30, 45, 60, 75, 90, 105, 120, 135, 150],
 [16, 32, 48, 64, 80, 96, 112, 128, 144, 160],
 [17, 34, 51, 68, 85, 102, 119, 136, 153, 170],
 [18, 36, 54, 72, 90, 108, 126, 144, 162, 180],
 [19, 38, 57, 76, 95, 114, 133, 152, 171, 190],
 [20, 40, 60, 80, 100, 120, 140, 160, 180, 200]]

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>23. Use list comprehension to find transpose of the matrix given below.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

<table>
    <tr>
        <td>1</td>
        <td>2</td>
    </tr>
    <tr>
        <td>3</td>
        <td>4</td>
    </tr>
    <tr>
        <td>5</td>
        <td>6</td>
    </tr>
    <tr>
        <td>7</td>
        <td>8</td>
    </tr>
    </table>

In [43]:
lst = [[1,2],[3,4],[4,5],[5,6],[7,8]]

t = [[row[i] for row in lst] for i in range(2)]

print(t)

[[1, 3, 4, 5, 7], [2, 4, 5, 6, 8]]
